server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -1503,6 +1493,18 @@ public DeleteResult delete(Delete delete) throws IOException {
                         plan.currentlyDeleted == false
                     );
                 }
+                if (plan.deleteFromLucene) {
```

Comment:
Shouldn't it be moved into `if (plan.deleteFromLucene || plan.addStaleOpToLucene)` block? Not a big deal, but we already have a check for  `plan.deleteFromLucene` in there:

```
if (plan.deleteFromLucene || plan.addStaleOpToLucene) {
    deleteResult = deleteInLucene(delete, plan);
    if (plan.deleteFromLucene) {
     ....
    }
} ...
```

Reply:
Curious if the JVM already optimizes for that? Either way, I won't argue w/ saving the extra check :)

Reply:
It is more from readability perspective, the things which happen when `plan.deleteFromLucene ` are kept closer to each other, otherwise it is difficult to build a picture (at least when I read the code), thank you!